original/zip_image_search.py

```diff
@@ -0,0 +1,178 @@
+##Intro code--import libraries and modules, load face detection classifier, import the small file to with which to practice,
+##and then open the zip file and create a list of the files inside
+
+
+#     file_name_str = 'readonly/small_img.zip'
+#     face_cascade_str = 'readonly/haarcascade_frontalface_default.xml'
+
+def main (name_str, file_name_str, face_cascade_str):
+    
+    import zipfile
+
+    from PIL import Image
+    from PIL import ImageDraw
+    import pytesseract
+    import cv2 as cv
+    import numpy as np
+
+    file_name_str = file_name_str
+    face_cascade_str = face_cascade_str
+    
+    search_name_return_faces (name_str, file_name_str, face_cascade_str)
+
+#this function accepts a zipfile as a string and returns a list of the files, in this case, image files
+def create_list_unzipped_files(file_name_str):
+    print ("Starting to create_list_unzipped_files")
+   
+    # specifying the zip file name 
+    file_name = file_name_str
+  
+    # opening the zip file in READ mode 
+    with zipfile.ZipFile(file_name, 'r') as zip: 
+        # printing all the contents of the zip file 
+#         zip.printdir() 
+  
+        # extracting all the files 
+        print("Extracting all the files now...") 
+        zip.extractall()
+    
+    #create a list of the files within the ZipFile
+    list_unzipped_files = zip.namelist()
+   
+    print (("Number of files in zip folder is: ") + (str(len(zip.namelist()))))
+    
+    display (list_unzipped_files)
+    
+    return list_unzipped_files
+
+    print ("Done with create_list_unzipped_files!")
+
+# #this function takes a list [(x,y,w,h),...] and returns [(x1, y1, x2, y2),....] which is more usable for creating a 
+# bounding_box by which the image is cropped and the resulting images are saved to a list 
+def create_list_face_thmbnl_imgs(array_name, list_name):
+    print ("Starting to create_list_face_thmbnl_imgs")
+
+    # Lets create a PIL image object
+    pil_img=Image.fromarray(array_name,mode="L")
+    
+    # And lets create our drawing context
+    drawing=ImageDraw.Draw(pil_img)
+    
+    list_thmbnl_images = [ ]
+    
+    num = 0 
+    
+    for xywh in list_name:
+        num = num + 1
+        
+    # Now we just draw a rectangle around the bounds
+        x1 = xywh[0]
+        y1 = xywh[1]
+        x2 = xywh[0]+xywh[2]
+        y2 = xywh[1]+xywh[3]
+
+        bounding_box = (x1, y1, x2, y2)
+        
+        single_face=pil_img.crop(bounding_box)
+        
+        #display(single_face)
+    
+        single_face_copy = single_face.copy()
+        
+        single_face_copy.save('single_face_copy_thmbnl_{}.jpg'.format(num))
```

Comment:
Based on the rest of the code I believe this `save` call is not needed.

`single_face_copy` is appended to the `list_thmbnl_images` and the files do not appear to be loaded later